sofar/sofastream.py

```diff
@@ -0,0 +1,192 @@
+from netCDF4 import Dataset
+import numpy as np
+
+class SofaStream():
```

Comment:
```suggestion
class SofaStream(Sofa):
```
which doesnt it enherit from the sofa class, then it could share all functionalities. If I remember correctly, it should behave like a real Sofa class exept its reading from file and not from ram, and also the data is a netBla object if you dont slice. Then all the tests with sofa should also work with SofaStream.

Reply:
Not sure about this, since SofaStream is just imitating the attribute/variable-access and `inspect()` of the `Sofa` class.

Reply:
No I think you're right. Inheriting does't work here. The `Sofa` class holds all the data of a sofa-file while `SofaStream` reads it upon request. Due to this even the `inspect` function that is contained in both classes has partly different code.

It might be possible to do this but it would require quite a large code refactoring.

Reply:
From my side, the current implementation is fine considering that we now have a general introduction for the module that explains Sofa and SofaStream. In addition users would get an AttrubiteError if for example trying to call `add_attribute`.